app/controllers/sample_webpush_notifications_controller.rb

```diff
@@ -1,4 +1,6 @@
 class SampleWebpushNotificationsController < ApplicationController
+  before_action :require_logged_in
```

Comment:
I'm not sure if it's not good or not that blocking the `#create` request by redirecting. But it would be better than a crash to me.